File: server_client_java_rpc_protocol/JavaSocketsFestival_protocol_rpc/Java11ChatSockets/ChatPersistence/src/main/java/festival/persistence/repository/DBRepoBilet.java
package festival.persistence.repository;

import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import festival.model.Artist;
import festival.model.Bilet;
import festival.model.Identifier;
import festival.model.Spectecol;

import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Optional;
import java.util.Properties;

public class DBRepoBilet implements IDBRepoBilet{
    private JdbcUtils dbUtils;
    private static final Logger logger= LogManager.getLogger();


    public DBRepoBilet(Properties props) {
        logger.info("Initializing DBRepoBilet with properties: {} ",props);
        dbUtils=new JdbcUtils(props);
    }


    @Override
    public Iterable findAll() {

        logger.traceEntry();
        Connection con = dbUtils.getConnection();
        List<Bilet> Bilete = new ArrayList<>();
        try (PreparedStatement preStmt = con.prepareStatement(
                "SELECT * \n" +
                        "FROM bilet  \n" ))
        {
            try (ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {

                    int id_spectacol = result.getInt("IDSpectacol");
                    int id_angajat= result.getInt("IDAngajat");
                    int id_client = result.getInt("IDClient");
                    int id_bilet = result.getInt("IDBilet");
                    int nrLocuri = result.getInt("numarLocuriB");


                    Bilet bilet= new Bilet((long) id_angajat, (long) id_client, (long) id_spectacol,nrLocuri);
                   bilet.setId((long) id_bilet);
                    Bilete.add(bilet);

                }
            }
        } catch (SQLException e) {
            logger.error(e);
            System.err.println("Error DB " + e);
        }
        logger.traceExit(Bilete);
        return Bilete;

    }

    @Override
    public Optional save(Identifier entity) {
        logger.traceEntry("saving task {} ", entity);
        Connection con = dbUtils.getConnection();
        try (PreparedStatement preStmt = con.prepareStatement("insert into bilet (IDAngajat,IDSpectacol,IDClient,numarLocuriB) values (?,?,?,?)")) {
            preStmt.setInt(1, ((Bilet)entity).getId_angajat().intValue());
            preStmt.setInt(2, ((Bilet)entity).getId_spectacol().intValue());
            preStmt.setInt(3, ((Bilet)entity).getId_client().intValue());
            preStmt.setInt(4,  ((Bilet)entity).getNr_locuri());
            int result = preStmt.executeUpdate();
            logger.trace("Saved {} instances", result);
        } catch (SQLException ex) {
            logger.error(ex);
            System.err.println("Error DB " + ex);
        }
        logger.traceExit();
        return Optional.ofNullable(entity);
    }

    @Override
    public Optional findOne(Object longID) {
        return Optional.empty();
    }

    @Override
    public Optional delete(Object aLong) {
        return Optional.empty();
    }


}

File: server_client_C_sharp/mpp_c_app1/FestivalPersistence/DBRepoBilet.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Data.SQLite;
using FestivalModel;


namespace FestivalPersistence
{


    public interface IDBRepoBilet : IRepository<long, Bilet>
    {
    }

    public class DBRepoBilet : IDBRepoBilet
    {
        private SQLiteConnection connection;
        private static readonly ILog logger = LogManager.GetLogger("DBRepoBilet");

        public DBRepoBilet(string connectionString)
        {
            logger.Info("Initializing DBRepoBilet...");
            connection = new SQLiteConnection(connectionString);
        }

        public Bilet findOne(long id)
        {
            throw new NotImplementedException();
        }

        public IEnumerable<Bilet> findAll()
        {
            logger.InfoFormat("findAll method called.");
            List<Bilet> Bilete = new List<Bilet>();
            using (SQLiteCommand command = new SQLiteCommand("SELECT * FROM bilet", connection))
            {
                connection.Open();
                using (SQLiteDataReader reader = command.ExecuteReader())
                {
                    while (reader.Read())
                    {
                        int id_spectacol = reader.GetInt32(reader.GetOrdinal("IDSpectacol"));
                        int id_angajat = reader.GetInt32(reader.GetOrdinal("IDAngajat"));
                        int id_client = reader.GetInt32(reader.GetOrdinal("IDClient"));
                        int id_bilet = reader.GetInt32(reader.GetOrdinal("IDBilet"));
                        int nrLocuri = reader.GetInt32(reader.GetOrdinal("numarLocuriB"));

                        Bilet bilet = new Bilet(id_angajat, id_client, id_spectacol, nrLocuri);
                        bilet.IdBilet = id_bilet;
                        Bilete.Add(bilet);
                    }
                }

                connection.Close();
            }

            return Bilete;
        }

        public void save(Bilet entity)
        {
            logger.InfoFormat("saving task {0} ", entity);
            using (SQLiteCommand command =
                   new SQLiteCommand(
                       "INSERT INTO bilet (IDAngajat, IDSpectacol, IDClient, numarLocuriB) VALUES (@id_angajat, @id_spectacol, @id_client, @nr_locuri)",
                       connection))
            {
                command.Parameters.AddWithValue("@id_angajat", entity.IdAngajat);
                command.Parameters.AddWithValue("@id_spectacol", entity.IdSpectacol);
                command.Parameters.AddWithValue("@id_client", entity.IdClient);
                command.Parameters.AddWithValue("@nr_locuri", entity.NrLocuri);

                connection.Open();
                int result = command.ExecuteNonQuery();
                connection.Close();
                logger.InfoFormat("Saved {0} instances", result);
            }

        }

        public void delete(long id)
        {
            throw new NotImplementedException();
        }
    }
}